function/random/random.js

```diff
@@ -4,49 +4,63 @@ module.exports = function(RED) {
+           if (n.high !== "") {this.high  = parseFloat(n.high)}
+              else {this.high = n.high}; 
+      
+           if (node.low === "") {
+               node.low = 1;
+               if ('from' in msg) {node.low = msg.from} 
+           }
+                
+           if (node.high === "") {
+               node.high = 10; 
+               if ('to' in msg) {node.high = msg.to} 
+           }
+
+           // if returning an interger, do a parseInt() on the low and high values
+           if ( (node.inte == "true") || (node.inte === true) ) {
+               node.low  = parseInt(node.low);
+               node.high = parseInt(node.high);
+           }
 
-            if (node.low === "") {
-                node.low = 1
-                if ('from' in msg) {
-                    if ( (typeof msg.from === 'number') || ( (typeof msg.from === 'string') && (!isNaN(Number(msg.from)) ) )) {
-                        node.low = Number(msg.from)
-                    }
-                } 
-            }
-    
-            if (node.high === "") {
-                node.high = 10 
-                if ('to' in msg) {
-                    if ( (typeof msg.to === 'number') || ( (typeof msg.to === 'string') && (!isNaN(Number(msg.to)) ) )) {
-                        node.high = Number(msg.to)
-                    }
-                } 
-            }
+           // flip the values if low > high so random will work
+           var value;
+           if (node.low > node.high) {  
+               value = node.low
+               node.low = node.high
+               node.high = value
+           }
+           // generate the random number
+           if ( (node.inte == "true") || (node.inte === true) ) {
+               node.low  = Math.round(node.low)
+               node.high = Math.round(node.high)
+               value = Math.round(Math.random() * (node.high - node.low + 1) + node.low - 0.5);
+           } else {
+               value = Math.random() * (node.high - node.low) + node.low;
+           }
             
-            // flip the values if low > high
-            var value;
-            if (node.low > node.high) {  
-                value = node.low
-                node.low = node.high
-                node.high = value
-            }
-
-            if (node.inte == "true" || node.inte === true) {
-                value = Math.round(Math.random() * (node.high - node.low + 1) + node.low - 0.5);
-            }
-            else {
-                value = Math.random() * (node.high - node.low) + node.low;
-            }
-            RED.util.setMessageProperty(msg,node.property,value);
-            node.send(msg);
+           RED.util.setMessageProperty(msg,node.property,value);
+
+           // output a status under the node
+           var color = "grey";
+           if ( (isNaN(node.low)) || (isNaN(node.high)) ) {
+              this.status({fill:"red",shape:"dot",text:"random: from " + node.low + " to " + node.high})
```

Comment:
Not sure we need status for this sort of thing - just increases traffic to front end. Yes we need the node.error

Reply:
status removed - I found it helpful in the editor when sending different from/to's to the node